pkg/auth/providers/saml/saml_provider.go

```diff
@@ -31,21 +34,26 @@ const (
 	ADFSName            = "adfs"
 	KeyCloakName        = "keycloak"
 	OKTAName            = "okta"
+	ShibbolethName      = "shibboleth"
 	loginAction         = "login"
 	testAndEnableAction = "testAndEnable"
 )
 
 type Provider struct {
-	ctx             context.Context
-	authConfigs     v3.AuthConfigInterface
-	secrets         corev1.SecretInterface
-	userMGR         user.Manager
-	tokenMGR        *tokens.Manager
-	serviceProvider *saml.ServiceProvider
-	name            string
-	userType        string
-	groupType       string
-	clientState     samlsp.ClientState
+	ctx                context.Context
+	authConfigs        v3.AuthConfigInterface
+	secrets            corev1.SecretInterface
+	userMGR            user.Manager
+	tokenMGR           *tokens.Manager
+	serviceProvider    *saml.ServiceProvider
+	name               string
+	userType           string
+	groupType          string
+	certs              string
```

Comment:
anything that is ldap specific, prefix it as such. But also, why arent the certs and caPool part of ldapProvider?